src/metabase/sync/sync_metadata/fks.clj

```diff
@@ -106,27 +122,32 @@
-       {:total-fks   (count fks-to-update)
-        :updated-fks (sync-util/sum-numbers (fn [fk]
-                                              (mark-fk! database table fk))
-                                            fks-to-update)}))))
+     (let [fk-metadata (fetch-metadata/table-fk-metadata database table)]
+       {:total-fks   (count fk-metadata)
+        :updated-fks (sync-util/sum-numbers #(mark-fk! database %) fk-metadata)}))))
 
 (mu/defn sync-fks!
   "Sync the foreign keys in a `database`. This sets appropriate values for relevant Fields in the Metabase application
-  DB based on values from the `FKMetadata` returned by [[metabase.driver/describe-table-fks]]."
+  DB based on values from the `FKMetadata` returned by [[metabase.driver/describe-table-fks]].
+
+  If the driver supports the `:describe-fks` feature, [[metabase.driver/describe-fks]] is used to fetch the FK metadata.
+
+  This function also sets all the tables that should be synced to have `initial-sync-status=complete` once the sync is done."
   [database :- i/DatabaseInstance]
-  (reduce (fn [update-info table]
-            (let [table         (t2.realize/realize table)
-                  table-fk-info (sync-fks-for-table! database table)]
-              ;; Mark the table as done with its initial sync once this step is done even if it failed, because only
-              ;; sync-aborting errors should be surfaced to the UI (see
-              ;; `:metabase.sync.util/exception-classes-not-to-retry`).
-              (sync-util/set-initial-table-sync-complete! table)
-              (if (instance? Exception table-fk-info)
-                (update update-info :total-failed inc)
-                (merge-with + update-info table-fk-info))))
-          {:total-fks    0
-           :updated-fks  0
-           :total-failed 0}
-          (sync-util/db->reducible-sync-tables database)))
+  (u/prog1 (if (driver/database-supports? (driver.u/database->driver database) :describe-fks database)
+             (sync-fks-for-db! database)
+             (reduce (fn [update-info table]
+                       (let [table         (t2.realize/realize table)
+                             table-fk-info (sync-fks-for-table! database table)]
+                         (sync-util/set-initial-table-sync-complete! table)
+                         (if (instance? Exception table-fk-info)
+                           (update update-info :total-failed inc)
+                           (merge-with + update-info table-fk-info))))
+                     {:total-fks    0
+                      :updated-fks  0
+                      :total-failed 0}
+                     (sync-util/db->reducible-sync-tables database)))
+    ;; Mark the table as done with its initial sync once this step is done even if it failed, because only
+    ;; sync-aborting errors should be surfaced to the UI (see
+    ;; `:metabase.sync.util/exception-classes-not-to-retry`).
+    (sync-util/set-initial-table-sync-complete-for-db! database)))
```

Comment:
It's not related to this PR, but I'm surprised to see this is done in `sync-fks!` as it has nothing to do with fk. maybe it should be its own sync step.